src/md.rs

```diff
@@ -235,33 +479,40 @@ struct SuperWeirdWordPlugin {
     foreign_long_watch: bool,
     bad_character_count: u64,
     buffer_accent_count: u64,
-    buffer: String,
+    buffer: Vec<MessDetectorChar>,
 }
 
 impl MessDetectorPlugin for SuperWeirdWordPlugin {
-    fn eligible(&self, character: char) -> bool {
+    fn eligible(&self, character: &MessDetectorChar) -> bool {
         true
     }
-    fn feed(&mut self, character: char) {
-        if character.is_ascii_alphabetic() {
-            self.buffer.push(character);
-            if is_accentuated(character) {
+    fn feed(&mut self, character: &MessDetectorChar) {
+        if character.is(MessDetectorCharFlags::ASCII_ALPHABETIC) {
+            self.buffer.push(*character);
+            if character.is(MessDetectorCharFlags::ACCENTUATED) {
                 self.buffer_accent_count += 1;
             }
-            self.foreign_long_watch |= (!is_latin(character) || is_accentuated(character))
-                && !is_cjk(character)
-                && !is_hangul(character)
-                && !is_katakana(character)
-                && !is_hiragana(character)
-                && !is_thai(character);
-
+            if !self.foreign_long_watch
```

Comment:
can be rewritten like this?
```
foreign_long_watch |= (!character.is(MessDetectorCharFlags::LATIN) || (character.is(MessDetectorCharFlags::ACCENTUATED))
                    && !character.is(MessDetectorCharFlags::CJK)
                    && !character.is(MessDetectorCharFlags::HANGUL)
                    && !character.is(MessDetectorCharFlags::KATAKANA)
                    && !character.is(MessDetectorCharFlags::HIRAGANA)
                    && !character.is(MessDetectorCharFlags::THAI)
```

Reply:
Yes, sure. I didn't think about idiomatic code at all - my plan was to do it and see new performance and after that moment rewrite in more idiomatic way with performance controlling :)